resources/clj-kondo.exports/tensegritics/clojuredart/hooks/flutter.clj

```diff
@@ -0,0 +1,42 @@
+(ns hooks.flutter
+  (:require [clj-kondo.hooks-api :as api]
+            [clojure.set :as set]
+            [clojure.string :as str]))
+
+(defn widget 
+  "Hook for cljd.flutter.alpha/widget macro"
+  [{:keys [:node]}]
+    (let [args (rest (:children node))
+          opts-nodes (take-while (comp api/keyword-node? first) (partition 2 args))
+          [_ state] (first (filter #(-> % first (api/sexpr) (= :state)) opts-nodes))
+
+          bindings-count (* 2 (count opts-nodes))
+          body (drop bindings-count args)
+          error (fn [msg] (throw (ex-info msg {})))
+          unknown-keys (set/difference 
+                         (into #{} (map (comp api/sexpr first) opts-nodes))
+                         #{:state :key :watch :context})]
+
+      (cond
+        (and state (not (api/vector-node? state)))
+        (error "State should be a vector [name initial-value]")
+
+        (and state (not= (count (:children state)) 2))
+        (error "State should contain only one name-value pair")
+
+        (seq unknown-keys)
+        (error (str "Unknown keys " (str/join " " unknown-keys)))
+
+        (not= 1 (- (count args) bindings-count)) ;; body is the one that remains 
+        (error (str "Unknown values"))
+
+        :else 
+        (let [check-symbols #{:watch :key}
```

Comment:
I don't understand what's this function must return. There seems to be keywords in binding position in the generated let.

Reply:
Now it returns something like:

```clojure
(let [context-name-user-provided identity ; if user privides context
      state-name state-value ; if user provides a state 
      ]

    ;; need it to check if these symbols exist 
    watch ; could be nil
    key ; could be nil 

    (widget. (body. (creation. )))
```